lib/ts/util/itinerary.ex

```diff
@@ -223,6 +227,7 @@ defmodule Util.Itinerary do
   # Signals iteration to stop when schedule is empty or
   # all neighbours have been visited
   defp stop_fn(neighbours, schedule) do
+    Logger.debug(fn -> "module=Util.Itinerary; message=Log message" end)
```

Comment:
Log message = all neighbours visited.

Reply:
A few other modifications:
* debug statement for value of should_stop on line-81
* debug statement for return value from _find_valid_connection(vars) on between lines-83 and 84